### PR TITLE
complete chapter9

### PR DESCRIPTION
**Pull request by vvtrung**
image

### CODE REVIEW

**Comment by vvtrung** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user && user.authenticate(params[:session][:password])
-      log_in user
-      redirect_to user
+    if @user && @user.authenticate(params[:session][:password])
+      log_in @user
+      forget(@user)
```
Comment (hhhhuy112) [inline app/controllers/sessions_controller.rb]: remove ()

**Comment by hhhhuy112** [comment]
commented

**Comment by vvtrung** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -1,17 +1,44 @@
...
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update(remember_digest: User.digest(remember_token))
+  end
+
+  # Returns true if the given token matches the digest.
+  def authenticated? remember_token
+    return false if remember_digest.nil?
```
Comment (hhhhuy112) [inline app/models/user.rb]: remember_digest.blank?

**Comment by hhhhuy112** [comment]
commented

**Comment by vvtrung** [comment]
ready

**Review by hhhhuy112 — APPROVED** [review]

**Comment by hhhhuy112** [comment]
merged
Mợt đi em

**Pull request merged**